homeassistant/components/wallbox/switch.py

```diff
@@ -0,0 +1,88 @@
+"""Home Assistant component for accessing the Wallbox Portal API. The lock component creates a switch entity."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Any
+
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import WallboxCoordinator, WallboxEntity
+from .const import (
+    CHARGER_DATA_KEY,
+    CHARGER_PAUSE_RESUME_KEY,
+    CHARGER_SERIAL_NUMBER_KEY,
+    CHARGER_STATUS_DESCRIPTION_KEY,
+    DOMAIN,
+)
+
+
+@dataclass
+class WallboxSwitchEntityDescription(SwitchEntityDescription):
+    """Describes Wallbox sensor entity."""
+
+
+SWITCH_TYPES: dict[str, WallboxSwitchEntityDescription] = {
+    CHARGER_PAUSE_RESUME_KEY: WallboxSwitchEntityDescription(
+        key=CHARGER_PAUSE_RESUME_KEY, name="Pause/Resume"
+    ),
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Create wallbox sensor entities in HASS."""
+    coordinator: WallboxCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        [WallboxSwitch(coordinator, entry, SWITCH_TYPES[CHARGER_PAUSE_RESUME_KEY])]
+    )
+
+
+class WallboxSwitch(WallboxEntity, SwitchEntity):
+    """Representation of the Wallbox portal."""
+
+    entity_description: WallboxSwitchEntityDescription
+    coordinator: WallboxCoordinator
+
+    def __init__(
+        self,
+        coordinator: WallboxCoordinator,
+        entry: ConfigEntry,
+        description: WallboxSwitchEntityDescription,
+    ) -> None:
+        """Initialize a Wallbox switch."""
+
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._coordinator = coordinator
```

Comment:
```suggestion
```
The coordinator is already assigned to `self.coordinator` with the `super()` call.